### ISSUE TITLE
"Show More {facet_type}" (Tags, Formats, Groups, Organizations) labels in sidebars can be mistranslated (French+)

### ISSUE THREAD

**Issue by Chealer** [ckan/ckan#5332]
### Please describe the expected behaviour
Pages such as /dataset contain facet boxes in their left sidebar. When a certain facet contains more items than a certain limit, a link offers controlling the display of surplus items. For example, when the Organizations facet contains more than 10 organizations, it initially shows a "Show More Organizations" link under the tenth organization. If that link is clicked, the target page offers to cancel with a "Show Only Popular Organizations" link instead.

When using CKAN in a language other than English, the labels for these links should be appropriately translated. For example, in French, "Show More Organizations" should become "Montrer plus d'organisations".

### Please describe the actual behaviour
Depending on the facet and on the language, label translations may be incorrect. For example, in French, "Show More Organizations" becomes "Montrer plus d**e O**rganisations".

### What steps can be taken to reproduce the issue? 
1. Create more organisations than the limit.
1. Set CKAN in French
1. Go to /dataset

---

I am not polyglot enough to tell all the problems with the current approach, but the case of French shows 2:

1. Capitalization; As mentioned in ticket #302 the label's capitalization is probably sub-optimal even in English, but having title capitalization in labels is not rare there. But in French, capitalizing more than the first letter is totally incorrect.
2. [Elision](https://en.wikipedia.org/wiki/Elision_(French)); In French, it is not possible to translate "Show more _some_noun_" to a single string. In general, the translation would be "Montrer plus de _some_noun_", as is currently the case, but if _some_noun_ starts with a vowel, the translation would be "Montrer plus d'_some_noun_", since "de" requires an apostrophe in front of a vowel. This is a counter-example of the "But don’t split a string into separate strings." guideline documented in [the CKAN Contributing guide's General guidelines for internationalizing strings](https://docs.ckan.org/en/2.8/contributing/string-i18n.html#general-guidelines-for-internationalizing-strings).

The second issue is worst. I think it only creates a problem for organizations, but that translation looks ridiculous.

French is a tricky language, but there are probably many more issues in other languages, as French is close to English.

---

For French, the capitalization issue could be solved by making facet_type lowercase.
The elision issue is unfortunately more complicated. To my knowledge, gettext does not offer a mechanism to deal with elision, and the only safe way would be to have a specific string for "Show more organisations".

**Comment by Chealer** [comment]
We worked around this with a small enough change. The following is a diff against a custom template, so it won't apply cleanly on CKAN 2.8.3, this is just to give an idea:
```
diff --git a/ckanext/dqc_scheming/i18n/fr/LC_MESSAGES/ckanext-dqc_scheming.po b/ckanext/dqc_scheming/i18n/fr/LC_MESSAGES/ckanext-dqc_scheming.po
index 023a68f..c2393c7 100644
--- a/ckanext/dqc_scheming/i18n/fr/LC_MESSAGES/ckanext-dqc_scheming.po
+++ b/ckanext/dqc_scheming/i18n/fr/LC_MESSAGES/ckanext-dqc_scheming.po
@@ -110,3 +103,6 @@ msgstr "Non-respect des Lignes directrices au niveau des nombres, "
 msgid "Guidelines conformity for boolean values not applied"
 msgstr "Non-respect des Lignes directrices au niveau des valeurs booléennes"
 
+#: ckanext/dqc_scheming/templates/snippets/facet_list.html
+msgid "Show more organizations"
+msgstr "Montrer plus d'organisations"
diff --git a/ckanext/dqc_scheming/templates/snippets/facet_list.html b/ckanext/dqc_scheming/templates/snippets/facet_list.html
index 1b9435e..07532c3 100644
--- a/ckanext/dqc_scheming/templates/snippets/facet_list.html
+++ b/ckanext/dqc_scheming/templates/snippets/facet_list.html
@@ -1,8 +1,10 @@
 {% block facet_list_items %}
   {% with items = items or h.get_facet_items_dict(name) %}
+  {% set noun = title.lower() %}
   {% if items %}
     <nav>
       <ul class="{{ nav_class or 'list-unstyled nav nav-simple nav-facet' }}">
@@ -36,14 +38,14 @@
     <p class="module-footer">
       {% if h.get_param_int('_%s_limit' % name) %}
         {% if h.has_more_facets(name) %}
-          <a href="{{ h.remove_url_param('_%s_limit' % name, replace=0, extras=extras, alternative_url=alternative_url) }}" class="read-more">{{ _('Show More {facet_type}').format(facet_type=title) }}</a>
+          <a href="{{ h.remove_url_param('_%s_limit' % name, replace=0, extras=extras, alternative_url=alternative_url) }}" class="read-more">{% if noun == 'organisations' %}{{ _('Show more organizations') }}{% else %}{{ _('Show More {facet_type}').format(facet_type=noun) }}{% endif %}</a>
         {% endif %}
       {% else %}
-        <a href="{{ h.remove_url_param('_%s_limit' % name, extras=extras, alternative_url=alternative_url) }}" class="read-more">{{ _('Show Only Popular {facet_type}').format(facet_type=title) }}</a>
+        <a href="{{ h.remove_url_param('_%s_limit' % name, extras=extras, alternative_url=alternative_url) }}" class="read-more">{{ _('Show Only Popular {facet_type}').format(facet_type=noun) }}</a>
       {% endif %}
     </p>
   {% else %}
-    <p class="module-content empty">{{ _('There are no {facet_type} that match this search').format(facet_type=title) }}</p>
+    <p class="module-content empty">{{ _('There are no {facet_type} that match this search').format(facet_type=noun) }}</p>
   {% endif %}
   {% endwith %}
 {% endblock %}
```

And of course, there's a much simpler approach, which is to use vaguer labels. In the context of organizations (for example), it would be quite evident that just "Show more" would mean show more organizations. This could be compensated by clarifying the role of these labels by prepending plus (+) icons.

If this is still considered undesirable, and we don't want to hurt languages without this problem, this could be solved by translating "Show More {facet_type}" to a constant string for problematic languages, even though that might be unclean (just "En montrer plus" for French, for example).

The clarity penalty would probably be much higher for the string "There are no {facet_type} that match this search", unfortunately.

**Comment by Chealer** [comment]
> And of course, there's a much simpler approach, which is to use vaguer labels. In the context of organizations (for example), it would be quite evident that just "Show more" would mean show more organizations. This could be compensated by clarifying the role of these labels by prepending plus (+) icons.

For what it's worth, I just realized that [the production instance of the site I'm working on](https://www.donneesquebec.ca/recherche/fr/dataset?_groups_limit=0) has had such +/- icons for years.

**Comment by amercader** [comment]
I agree that these strings are problematic and I had to customize them myself in the past. My suggestion would be to simplify them in core to "Show more" and "Show less"

**Comment by Chealer** [comment]
One additional consideration regarding the last case (There are no {facet_type} that match this search).

- A period is missing.
- This looks a bit silly when no search was done. When setting up CKAN, before the database is filled, such messages show on /dataset.

**Comment by Gauravp-NEC** [comment]
@amercader , I am interested to work on this issue. Shall I proceed with your suggestion of simplifying them in core to "Show more" and "Show less". Please let me know your opinion.
